src/org/joml/AABBi.java

```diff
@@ -0,0 +1,778 @@
+     *
+     * @param a
+     *          the x factor in the plane equation
+     * @param b
+     *          the y factor in the plane equation
+     * @param c
+     *          the z factor in the plane equation
+     * @param d
+     *          the constant in the plane equation
+     * @return <code>true</code> iff the plane intersects this AABB; <code>false</code> otherwise
+     */
+    public boolean testPlane(float a, float b, float c, float d) {
+        return Intersectionf.testAabPlane(minX, minY, minZ, maxX, maxY, maxZ, a, b, c, d);
+    }
+
+    /**
+     * Test whether the given plane intersects this AABB.
+     * <p>
+     * Reference: <a href="http://www.lighthouse3d.com/tutorials/view-frustum-culling/geometric-approach-testing-boxes-ii/">http://www.lighthouse3d.com</a> ("Geometric Approach - Testing Boxes II")
+     *
+     * @param plane
+     *          the plane
+     * @return <code>true</code> iff the plane intersects this AABB; <code>false</code> otherwise
+     */
+    public boolean testPlane(Planef plane) {
+        return Intersectionf.testAabPlane(this, plane);
+    }
+
+    /**
+     * Test whether <code>this</code> and <code>other</code> intersect.
+     *
+     * @param other
+     *          the other AABB
+     * @return <code>true</code> iff both AABBs intersect; <code>false</code> otherwise
+     */
+    public boolean testAABB(AABBi other) {
+        return this.maxX >= other.minX && this.maxY >= other.minY && this.maxZ >= other.minZ &&
+            this.minX <= other.maxX && this.minY <= other.maxY && this.minZ <= other.maxZ;
+    }
+
+    /**
+     * Test whether <code>this</code> and <code>other</code> intersect.
+     *
+     * @param other
+     *          the other AABB
+     * @return <code>true</code> iff both AABBs intersect; <code>false</code> otherwise
+     */
+    public boolean testAABB(AABBf other) {
+        return this.maxX >= other.minX && this.maxY >= other.minY && this.maxZ >= other.minZ &&
+            this.minX <= other.maxX && this.minY <= other.maxY && this.minZ <= other.maxZ;
+    }
+
+    /**
+     * Test whether this AABB intersects the given sphere with equation
+     * <code>(x - centerX)^2 + (y - centerY)^2 + (z - centerZ)^2 - radiusSquared = 0</code>.
+     * <p>
+     * Reference: <a href="http://stackoverflow.com/questions/4578967/cube-sphere-intersection-test#answer-4579069">http://stackoverflow.com</a>
+     *
+     * @param centerX
+     *          the x coordinate of the center of the sphere
+     * @param centerY
+     *          the y coordinate of the center of the sphere
+     * @param centerZ
+     *          the z coordinate of the center of the sphere
+     * @param radiusSquared
+     *          the square radius of the sphere
+     * @return <code>true</code> iff this AABB and the sphere intersect; <code>false</code> otherwise
+     */
+    public boolean testSphere(float centerX, float centerY, float centerZ, float radiusSquared) {
```

Comment:
You removed `testSphere(Spheref sphere)`. To keep the API consistent, you could've referenced this method in it.
```
    public boolean testSphere(Spheref sphere) {
        return testSphere(sphere.x, sphere.y, sphere.z, sphere.r * sphere.r);
    }
```